### PR TITLE
[WIP] Add sum as a static function

### PR DESCRIPTION
**Pull request by Syzygianinfern0**
## Description
Add `torch.sum` equivalent. This is to support "reducing across dimensions" (as @aanurraj required it :stuck_out_tongue: )

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [x] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

File: src/sympc/tensor/static.py [hunk 1]
```diff
@@ -118,6 +119,62 @@ def cat_share_tensor(session_uuid_str: str, *shares: Tuple[ShareTensor]) -> Shar
...
+
+    expected_shape = torch.sum(
+        torch.empty(tensor.shape), dim=dim, keepdim=keepdim
+    ).shape
+    result = MPCTensor(shares=sum_shares, session=session, shape=expected_shape)
+
+    return result
+
+
+def sum_share_tensor(
```
Comment (kamathhrishi) [inline src/sympc/tensor/static.py:155]: Could you extend this for RSTensor too?

File: tests/sympc/tensor/static_test.py [hunk 2]
```diff
@@ -141,3 +142,18 @@ def test_cat(get_clients):
     concatenated = cat([x, y])
 
     assert (secret_concatenated == concatenated.reconstruct()).all()
+
+
+@pytest.mark.parametrize("dim, keepdim", itertools.product([0, 1, 2], [True, False]))
+def test_sum(get_clients, dim, keepdim):
+    clients = get_clients(2)
+
+    x_secret = torch.arange(12).view(2, 2, 3)
```
Comment (kamathhrishi) [inline tests/sympc/tensor/static_test.py:151]: Add a step size something like 0.5 so that it represents decimal values too?

**Comment by gmuraru** [comment]
Q: Why can't we use the ```GradSum``` from ```grad_functions.py```

**Review by gmuraru — CHANGES_REQUESTED** [review]
See question

**Comment by Syzygianinfern0** [comment]
> Q: Why can't we use the `GradSum` from `grad_functions.py`

Yes, I noticed the same after making the PR. Although, for some reason passing the dim arg to it does not work right. I am investigating it. 

(I should have marked this as WIP but forgot to :sweat_smile: )

**Comment by gmuraru** [comment]
@Syzygianinfern0 could we close this?

**Comment by Syzygianinfern0** [comment]
> @Syzygianinfern0 could we close this?

Yep 😅
